### ISSUE TITLE
about the iou of results

### ISSUE THREAD

**Issue by elk-april** [zhusiling/UNets#1]
hello, I use my own data to train the model, however, there is little that can be done to improve the iou by tuning lr (currently the highest accuracy is 60%,lr=0.1/0.05/0.01, weight_decay=1e-8).
Is there any way to improve the performance?
looking forward to your reply！
(你好，我使用我自己的数据训练模型，但是，通过调整lr几乎无法改变提高iou（目前精度普遍最高在60%,分别尝试了0.1,0.05,0.01）。请问，有什么办法能提高性能吗？)

**Comment by elk-april** [comment]
I modified the structure of your code(divide to train/test)
and delete some codes,like below:
train.py
```
for x, y in tqdm(train_loader):
        x, y = x.to(device), y.to(device)

        input_images(x, y, i, n_iter, k)

        opt.zero_grad()

        y_pred = model_test(x)
        lossT = calc_loss(y_pred, y)     # Dice_loss Used
        train_loss += lossT.item() * x.size(0)
        lossT.backward()
        opt.step()
        x_size = lossT.item() * x.size(0)
        k = 2

    model_test.eval()
    torch.no_grad()

    for x1, y1 in tqdm(valid_loader):
        x1, y1 = x1.to(device), y1.to(device)

        y_pred1 = model_test(x1)
        lossL = calc_loss(y_pred1, y1)     # Dice_loss Used

        valid_loss += lossL.item() * x1.size(0)
        x_size1 = lossL.item() * x1.size(0)

    # print loss
    train_loss = train_loss / len(train_idx)
    valid_loss = valid_loss / len(valid_idx)

    if (i + 1) % 1 == 0:
        print('Validation loss decreased ({:.6f} --> {:.6f}).  Saving model '.format(valid_loss_min, valid_loss))
    torch.save(model_test.state_dict(),read_model_path+str(i+1)+".pth")
    loss_num.append(str(valid_loss)[0:5])

    print(read_model_path+str(i+1)+".pth  saved")

```
```
test.py
`for _ in tqdm(read_test_folder):

    # print(x_sort_test[i])
    im = Image.open(x_sort_test[i])
    im1 = im
    im_n = np.array(im1)
    im_n_flat = im_n.reshape(-1, 1)

    for j in range(im_n_flat.shape[0]):
        if im_n_flat[j] != 0:
            im_n_flat[j] = 255

    s = data_transform(im)
    pred = model_test(s.unsqueeze(0).cuda()).cpu()

    pred = F.sigmoid(pred)
    pred = pred.detach().numpy()
    pred = pred[0][0]

    img4 = (pred - pred.min()) * ((255 - 0) / (pred.max() - pred.min())) + 0
    img = img4.astype(np.uint8)
    threshold_predictions_v(img, args.thr)
    cv2.imwrite(res + x_sort_test[i].split('/')[-1], img)

    i = i + 1

imou = func.compute_mIoU(res, test_folderL)
print("checkpoints {} : miou　{}".format(args.num, imou))`
```
